src/chat/components/ChatMessagesList.tsx

```diff
@@ -10,16 +11,30 @@ function getMessagePosition(username: string) {
 
 export default function ChatMessagesList() {
   const data = useSelector(chatMessagesSelector);
+  let timeToCheck = new Date().toJSON().slice(0, 10);
+
   return (
     <Fragment>
-      {data.map((message: ChatMessage) => {
+      {data.map((message: ChatMessage, index) => {
+        const { id, created, text, author } = message;
+        const dataToDisplay = timeToCheck;
+        const messageDate = created.slice(0, 10);
+
+        const isLastMessage = index + 1 === data.length;
+        const isDifferentDate = messageDate !== timeToCheck;
+
         return (
-          <Message
-            key={message.id}
-            position={getMessagePosition(message.author)}
-            message={message.text}
-            created={message.created}
-          />
+          <Fragment key={id}>
+            {isDifferentDate && (timeToCheck = messageDate) && (
```

Comment:
Im not sure what is happening here: `timeToCheck = messageDate` . Is it a comparison or `timeToCheck` is being set to `messageDate`. If it is a comparison , then why not: timeToCheck === messageDate , or even further more pulling this out into a variable like: `check: boolean = timeToCheck === messageDate`

Reply:
It's not a comparision, we overwrite `timeToCheck` with the new date (it's how it works) 
It's in JSX, what may be confusing, but its purpose is to avoid checking `isDifferentDate === true ` twice, here in render and when we want to update `timeToCheck` variable

Reply:
I have put it a few lines higher to make it clearer